docs/CONTRIBUTING.md

```diff
@@ -102,7 +102,7 @@ python setup.py typecheck
 ```
 # CI Pipelines
 
-Every PR needs to pass CIs before merge. CI pipelines details are [here](CIPipelines.md).
+Every PR needs to pass CIs before merge. CI pipelines details are [here](CIPipelines.md). Passing [Lint Python CI](../.github/workflows/lint.yaml) is not required but recommended.
```

Comment:
Does this produce warnings about pre-existing code issues? Will it continue to do so after the other pending PRs to fix existing python code? If so, it would be worth explaining that here (something like "This may produce warnings about pre-existing code issues that are not required to be fixed in the PR. We plan to fix them in upcoming PRs".)

Reply:
There should be a way to apply it on git diff only.

Reply:
For now some checks will produce warnings about pre-existing code issues in "changed" files, which is sometimes confusing for PR review.

I assume setting `filter_mode: diff_context` will only applying it on git diff only. If that is true, we should set it for every check in https://github.com/onnx/onnx/blob/main/.github/workflows/lint.yaml. (like reviewdog/action-flake8, reviewdog/action-pyflakes, reviewdog/action-shellcheck) However, it seems to me that not every check supports filter_mode. For instance, [action-pyflakes](https://github.com/reviewdog/action-pyflakes). @justinchuby May I understand do you know how to only apply reviewdog/action-pyflakes on git diff only?

Reply:
That is precise. For flake8, since we enforce 0 errors in style.sh, it helps to expose all errors in the annotations. For shellcheck, since we don't usually run shellcheck locally, it helps to surface potential errors. pyflakes doesn't have the option. Other than that, I think the rest are diff_context

Reply:
For py flakes we may be able to either fork and update the check or create a pr to the check to enable this option

Reply:
We could also use "added", which will only show lints that are introduced

Reply:
After offline discussion, we concluded that for now we should use `filter_mode: diff_context` for every check. Since pyflake8 does not support it, we decided to remove it. For now the Python Lint CI should only review content from git diff. Thank @justinchuby for the quick update: https://github.com/onnx/onnx/pull/4371. Please review this PR again. Thank you all!